docs/platform_developers/developer_getting_started.rst

```diff
@@ -0,0 +1,194 @@
+********************************
+Developer Getting Started Guide
+********************************
+
+This guide explains how to set up the recommended developer setup on a Linux machine. 
+Other development setups are possible, but this one provides a good starting point.
+
+* Install VS Code and Inmanta extension.
+* Setting up Python virtual environments.
+* Benefit from linting and code navigation by setting up a project.
+* Set project sources
+* Module developers guide
+
+**The examples below are using ``pip`` your system might require you to use ``pip3``**.
+
+
+Install VS Code and Inmanta extension
+#######################################
+
+The developer setup is based on VSCode with the Inmanta extension.
+
+In order to install VS Code, you can refer to `this <https://code.visualstudio.com/learn/get-started/basics>`_ page.
+
+Inmanta's extension in VS Code marketplace can be found `here <https://marketplace.visualstudio.com/items?itemName=inmanta.inmanta>`_. 
+
+Further information about Inmanta VS Code extension is available on `this <https://github.com/inmanta/vscode-inmanta>`_ page.
+
+
+Setting up Python virtual environments
+########################################
+
+For every project that you work on, we recommend using a new virtual environment using ``venv``s. If you need a refresher, you can check out `this <https://docs.python.org/3/tutorial/venv.html>`_ page.
+
+To create a virtual environment:
+
+.. code-block:: bash
+    
+    python3 -m venv ~/.virtualenvs/my_project
+
+Then activate it by running:
+
+.. code-block:: bash
+    
+    source ~/.virtualenvs/my_project/bin/activate
+
+**Upgrading your ``pip`` will save you a lot of time and troubleshooting (due to changes in the pip resolver in version 20 and 21).** 
+
+You can do so by running:
+
+.. code-block:: bash
+    
+    pip install --upgrade pip
+
+
+Benefit from linting and code navigation by setting up a project
+##################################################################
+
+At the time of this writing, linting and code navigation in IDEs work only if you have a project, so even if you only work on a single module, it is best to have a project.
+
+There are two scenarios:
+
+1. Working on a new project :ref:`Working on a New Project`.
+2. Working on an existing project :ref:`Working on an Existing Project`.
+
+
+Working on a New Project
+========================
+
+To create a new project: 
+
+.. code-block:: bash
+
+    pip install cookiecutter
+
+    cookiecutter https://github.com/inmanta/inmanta-project-template.git
+
+
+For more details go `here <https://docs.inmanta.com/community/latest/model_developers/configurationmodel.html>`_.
+
+``cookiecutter`` is used to create projects in an easier and more convenient fashion.
```

Comment:
remove 80-82 as there is already a ref to the docs above.

Reply:
Done